internal/precache/workload/pullImages.go

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this inputFilePath except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package workload
+
+import (
+	"os"
+	"os/exec"
+	"strconv"
+	"sync"
+
+	"github.com/openshift-kni/lifecycle-agent/internal/precache"
+	log "github.com/sirupsen/logrus"
+)
+
+// MaxRetries is the max number of retries for pulling an image before marking it as failed
+const MaxRetries int = 5
+
+// Podman auth-file related constants
+const (
+	EnvAuthFile     string = "PULL_SECRET_PATH"
+	DefaultAuthFile string = "/var/lib/kubelet/config.json"
+)
+
+// executeCmd execute shell commands
+func executeCmd(suppressCmdOutput bool, cmd string, args ...string) (err error) {
```

Comment:
We have similar util commands, which have been recently refactored and moved to:
https://github.com/openshift-kni/lifecycle-agent/blob/main/ibu-imager/ops/execute.go

Can this be used instead? I guess the "suppression" option would need to be added

Reply:
It can be used, however, refactoring the function `func (e *executor) execute(liveLogger io.Writer, root, command string, args ...string) (string, error) ` in `ibu-imager/ops/execute.go` to include command output suppression should probably be done in a separate PR under its own Jira. 

Given that output suppression is not used elsewhere, it may be an unnecessary attempt resulting in extraneous parameter bloating for other use-cases. Since the `executeCmd` function is a simple wrapper to `exec/os.{Command,Run}`, I think it is okay to simply leave it here as is. If the command output suppression is beneficial to other scenarios, then a future refactoring may suffice.